nox/manifest.py

```diff
@@ -184,6 +189,30 @@ def make_session(
             func.should_warn[WARN_PYTHONS_IGNORED] = func.python
             func.python = False
 
+        if self._config.extra_pythons:
+            # If extra python is provided, expand the func.python list to
+            # include additional python interpreters
+            extra_pythons = self._config.extra_pythons  # type: List[str]
+            if isinstance(func.python, (list, tuple, set)):
+                func.python = _unique_list(*func.python, *extra_pythons)
+            elif not multi and func.python:
+                # If this is multi, but there is only a single interpreter, it
+                # is the reentrant case. The extra_python interpreter shouldn't
+                # be added in that case. If func.python is False, the session
+                # has no backend; treat None and the empty string equivalently.
+                # Otherwise, add the extra specified python.
+                assert isinstance(func.python, str)
+                func.python = _unique_list(func.python, *extra_pythons)
+            elif not multi:
+                # If no pythons are specified, consider extra_pythons to be an
+                # override, not an addition, as these sessions are often meant
+                # to be run once, not in a parameterized way.
+                if len(extra_pythons) == 1:
+                    # By setting func.python to a single str, the friendly name
+                    # will not have python appended, as there is a single run
+                    func.python = extra_pythons[0]
+                else:
+                    func.python = _unique_list(*extra_pythons)
```

Comment:
As proposed in my earlier comment, I would ignore sessions where python is None or False.

The suggestion below does this by deleting the `elif not multi:` branch.

```suggestion
        if self._config.extra_pythons:
            # If extra python is provided, expand the func.python list to
            # include additional python interpreters
            extra_pythons = self._config.extra_pythons  # type: List[str]
            if isinstance(func.python, (list, tuple, set)):
                func.python = _unique_list(*func.python, *extra_pythons)
            elif not multi and func.python:
                # If this is multi, but there is only a single interpreter, it
                # is the reentrant case. The extra_python interpreter shouldn't
                # be added in that case. If func.python is False, the session
                # has no backend; if None, it uses the same interpreter as Nox.
                # Otherwise, add the extra specified python.
                assert isinstance(func.python, str)
                func.python = _unique_list(func.python, *extra_pythons)

```